plugins/tugboat/access_token.go

```diff
@@ -0,0 +1,77 @@
+package tugboat
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AccessToken,
+		DocsURL:       sdk.URL("https://docs.tugboatqa.com/tugboat-cli/set-an-access-token/"),
+		ManagementURL: sdk.URL("https://dashboard.tugboatqa.com/access-tokens"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Tugboat.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.TempFile(tugboatConfig, provision.AtFixedPath("~/.tugboat.yml")),
```

Comment:
One downside of using the file provisioner with a fixed path is that it users won't be able to test drive this plugin without overwriting their existing configuration.

Seems that Tugboat also reads from the [`TUGBOAT_API_TOKEN` environment variable](https://github.com/TugboatQA/docker-tugboat-cli#read-the-token-using-an-environment-variable), so if that works well then that may be the more suitable option.

Reply:
@florisvdg I was having trouble finding that documentation; thanks for sharing! Updating with your suggestion.